### ISSUE TITLE
How to reduce GPU memory useage for YOLO11 training?

### ISSUE THREAD

**Issue by xhuan8** [ultralytics/ultralytics#17488]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.


### Question

I try to upgrade from YOLOv5 to YOLO11, but YOLO11 takes far more GPU memory than YOLOv5.
See the list below, YOLO11 uses double sized GPU memory, when image size increases the memory usage grows fast.
Original image size is 5472 * 1216, YOLOv5s takes 24GB for training, but YOLO11 crashes with OOM.

Model	Image size	Memory(GB)	Epoch Time(s)
YOLO11s	2736 * 608	8.76	                 8
YOLOv5s	2736 * 608	4.52	                 27
YOLO11n	2736 * 608	4.63	                 6
YOLOv5n	2736 * 608	2.33	                 26
YOLO11n	4104 * 912	16.4	                 13
YOLOv5n	4104 * 912	5.71	                 39










































































































































































































































































































































































































































































































































































































































































































































































































### Additional

_No response_

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xhuan8, thank you for reaching out about YOLO11's GPU memory usage 🚀! This is an automated response to assist you, but don't worry, an Ultralytics engineer will join the conversation soon.

We understand the challenges with GPU memory, especially when transitioning between models. Here are a few steps that might help:

1. **Minimum Reproducible Example**: Please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) demonstrating the issue. This will assist our team in diagnosing the problem effectively.
2. **Documentation**: You might find useful tips on managing memory usage in our comprehensive [Docs](https://docs.ultralytics.com), which include many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.
3. **Training Configurations**: For optimized training, confirm that you've followed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our Ultralytics community for broader support:
- You can chat in real-time with others on our [Discord](https://ultralytics.com/discord) 🎧.
- Engage in in-depth discussions on [Discourse](https://community.ultralytics.com).
- Share insights and solutions on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are using the latest version of `ultralytics` alongside all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This might resolve some issues:

```bash
pip install -U ultralytics
```

## Verified Environments

Run YOLO in these verified environments for optimal results:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Check the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon** AMI: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker**: Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the CI badge is green, this indicates all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, confirming the functionality of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across various platforms. 

An Ultralytics engineer will review your issue shortly. Thanks for your patience and contributions!

**Comment by Skillnoob** [comment]
You can enable autobatch by adding `batch=-1` to your train arguments. If you want more details on what it does and what other options there are for batch, I reccomend reading the [train argument docs](https://docs.ultralytics.com/modes/train/#train-settings).

**Comment by xhuan8** [comment]
> You can enable autobatch by adding `batch=-1` to your train arguments. If you want more details on what it does and what other options there are for batch, I reccomend reading the [train argument docs](https://docs.ultralytics.com/modes/train/#train-settings).

thanks for your replay, batch already set to 1.
It looks like YOLO can only process square image, border added to image (2736 * 608)
```
Params      GFLOPs  GPU_mem (GB)  forward (ms) backward (ms)                   input                  output
     9428566       398.5         8.636         110.1         358.6      (1, 3, 2752, 2752)                    list
     9428566       796.9        17.677          1154          7175      (2, 3, 2752, 2752)                    list
```

**Comment by Skillnoob** [comment]
Yes, only square image training is supported. Ultralytics automatically scales/pads the images and preserves the aspect ratio. Can you share what code you used to train the model?

**Comment by xhuan8** [comment]
@Skillnoob 
4 times of GPU memory for 1.5 times of image size
Model Image size Memory(GB)
YOLO11n 2736     4.63
YOLO11n 4104     16.4

```
from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO("yolo11s.yaml")
    model.train(data=r"E:\Dataset\TSIEdge\dataset.yaml", epochs=3, imgsz=2736, batch=1, workers=0, mosaic=0.0)
```

**Comment by Y-T-G** [comment]
It's not 1.5 times.

4104²/2736² = 2.25

**Comment by Skillnoob** [comment]
Pixel count doesn't scale linearly with image resolution. It scales in a quadratic manner. I would recommend you use a smaller `imsgz` and instead use something like SAHI to slice your images and avoid the need of tiny batch sizes and lots of VRAM.
